add unit detection for XRT data (#6725)
sunpy/sunpy: changelog/6725.bugfix.rst

> XRT Maps now detect and include the unit property.

Reviewer: ```suggestion
`.XRTMap` now will set the units for level 0 and level 1 files correctly.
```

Reviewer 2: This unit checking should work for any processing level, not just 0 or 1. Also, I don't think that intersphinx link is right.

> `~sunpy.map.sources.XRTMap` will now set the unit for all files correctly.

Reviewer 2: ```suggestion
`~sunpy.map.sources.XRTMap` will now set the unit for XRT files if the ``BUNIT`` key is missing.
```

> `~sunpy.map.sources.XRTMap` will now set the units for all files correctly.

Reviewer: ```suggestion
`~sunpy.map.sources.XRTMap` will now set the unit for all files correctly.
```

> `~sunpy.map.sources.XRTMap`` will now set the units for all files correctly.

Reviewer: ```suggestion
`~sunpy.map.sources.XRTMap` will now set the units for all files correctly.
```
